go-controller/pkg/ovn/master.go

```diff
@@ -471,28 +558,53 @@ func (oc *DefaultNetworkController) deleteStaleNodeChassis(node *kapi.Node) erro
 }
 
 // cleanupNodeResources deletes the node resources from the OVN Northbound database
-func (oc *DefaultNetworkController) cleanupNodeResources(nodeName string) error {
-	if err := oc.deleteNodeLogicalNetwork(nodeName); err != nil {
-		return fmt.Errorf("error deleting node %s logical network: %v", nodeName, err)
+func (oc *DefaultNetworkController) cleanupNodeResources(node *kapi.Node) error {
+	v4Mgmt, v6Mgmt, err := oc.getManagementPortIPs(node)
+	if err != nil {
+		return fmt.Errorf("failed to clean up host access route policies via mp0 for node %q, error: %w", node.Name, err)
+	}
+	var mgmtIPs []net.IP
+
+	if v4Mgmt != nil {
+		mgmtIPs = append(mgmtIPs, v4Mgmt)
+	}
+	if v6Mgmt != nil {
+		mgmtIPs = append(mgmtIPs, v6Mgmt)
+	}
+
+	// cleanup route policy towards mp0
+	if len(mgmtIPs) > 0 {
+		oc.policyRouteCleanup(mgmtIPs)
 	}
 
-	if err := oc.gatewayCleanup(nodeName); err != nil {
-		return fmt.Errorf("failed to clean up node %s gateway: (%w)", nodeName, err)
+	if err := oc.deleteNodeLogicalNetwork(node.Name); err != nil {
+		return fmt.Errorf("error deleting node %s logical network: %v", node.Name, err)
+	}
+
+	if err := oc.gatewayCleanup(node.Name); err != nil {
+		return fmt.Errorf("failed to clean up node %s gateway: (%w)", node.Name, err)
+	}
+
+	// cleanup node Address set
+	addrSetName := fmt.Sprintf("%s-node-ips", node.Name)
```

Comment:
you don't really need node-ips suffix here, as it should be implied from the address set type